app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,6 @@
+class RecipesController < ApplicationController
+  def show
+    @recipe = Recipe.find(params[:id])
+    @recipe_ingredients = Ingredient.all
```

Comment:
Your instance variable `@recipe_ingredients` is set to `Ingredient.all`, which is actually pulling ALL records from the `ingredients` table, and _not_ the ones that belong to this recipe. For that, you should use an association on Recipe, like: 
```
@recipe.ingredients
```

Depending on how you used this instance variable, it could be a problem.